fetchai/ledger/api/__init__.py

```diff
@@ -89,8 +98,10 @@ def sync(self, txs: Transactions, timeout=None):
                 failures = ['{}:{}'.format(tx_status.digest_hex, tx_status.status) \
                             for tx_status in failed_this_round]
                 raise RuntimeError('Some transactions have failed: {}'.format(', '.join(failures)))
-
+            now = datetime.now()
             successful_this_round = [status for status in remaining_statuses if status.successful]
+            successful_this_round = [status for status in successful_this_round if (now-_get_or_set_default_time(hold_times, status.digest_hex, now))>=hold_state]
+            hold_times.update({status.digest_hex: -1 for status in remaining_statuses if status.non_terminal})
```

Comment:
This line appears to have no effect - the behavior of _get_or_set_default_time is identical when a key has value -1 as when a key is missing. The only time this would have an affect is if a transactions state changes from successful to non-terminal, but that shouldn't be possible?

Reply:
yes exactly. Non-terminal state is possible when the block is dropped.